source/agora/node/Ledger.d

```diff
@@ -268,14 +272,14 @@ public class Ledger
         };
         serializePart(block, dg);
 
-         string file_path = format("%s/%s", dir, "index.dat");
-        scope File index_fd = File(file_path, "a");
+        string file_path = format("%s/%s", dir, "index.dat");
+        File index_file = File(file_path, "a");
```

Comment:
It does not make sense to me to get a `File` as an argument and then build the index file in this function. Instead both `File` should be passed, otherwise you have mixed responsibilities.

Reply:
added index_File param